src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
Thanks for the benchmark results. For increased redability and reduced complexity (and all the maintenance and security benefits that come with it), I'm in favor of using `URI`. A 1.64x slowdown compared to "max" seems totally okay to me. In terms of Moore's Law, this performance "issue" will be amortized in less than a year. ;-)